jbx-protocol/juice-interface
# Show error message when payment fails

## What does this PR do and why?
- Add an onError callback to the transaction options object. I took this suggestion from @aeolianeth in #634 after doing a little bit of investigation.

- Add a notif for the end user when the transaction is missing required parameters.



#### How to test:
- Add a true boolean to `PayV1ProjectTx.ts` hook where the new `onError` function is introduced.
- Click the payment button in the v1 payment modal (shown in gif) in `http://localhost:3000/#/p/<project>`.
- error message should be shown to user.

## Screenshots or screen recordings
Gif of functionality:
<img src="https://user-images.githubusercontent.com/2502947/162634683-701c4071-5591-4dc6-b1d0-eb80401f2b48.gif" height="400px" />


## Acceptance checklist
